File: python/ejercicios_jp/crud-json-db/app/domain/model/persona.py
class Persona:
    def __init__(self, nombre: str, apellido: str, edad: int):
        self._nombre = nombre
        self._apellido = apellido
        self._edad = edad

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, nombre: str):
        self._nombre = nombre

    @property
    def apellido(self):
        return self._apellido

    @apellido.setter
    def apellido(self, apellido: str):
        self._apellido = apellido

    @property
    def edad(self):
        return self._edad

    @edad.setter
    def edad(self, edad: int):
        self._edad = edad

    def __str__(self):
        return f"Nombre: {self.nombre}, Apellido: {self.apellido}, Edad: {self.edad}"

File: programación/ut_9_gestion_de_datos_en_memoria/ejercicios/src/main/java/org/example/gestion_datos_en_memoria/ejercicios_jp/ejercicios_set/tree_set_implementado/Persona.java
package org.example.gestion_datos_en_memoria.ejercicios_jp.ejercicios_set.tree_set_implementado;

import java.util.Comparator;
import java.util.Objects;

public class Persona implements Comparable<Persona> {
    private String nombre;
    private int edad;

    public Persona(String nombre, int edad){
        this.nombre = nombre;
        this.edad = edad;
    }

    public String getNombre() {
        return nombre;
    }

    public int getEdad() {
        return edad;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Persona persona = (Persona) o;
        return Objects.equals(nombre, persona.nombre) && Objects.equals(edad, persona.edad);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nombre, edad);
    }

    @Override
    public String toString() {
        return "Persona{" +
                "nombre='" + nombre + '\'' +
                ", edad=" + edad +
                '}';
    }

    @Override
    public int compareTo(Persona otra) {
        int res = Integer.compare(this.edad, otra.edad);
        if (res == 0) {
            return this.nombre.compareTo(otra.nombre);
        }
        return res;
    }

}
